src/Http/Requests/QuestionnaireFrontAnswerRequest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace EscolaLms\Questionnaire\Http\Requests;
+
+use EscolaLms\Questionnaire\Models\Questionnaire;
+use EscolaLms\Questionnaire\Models\QuestionnaireModelType;
+use EscolaLms\Questionnaire\Rules\ClassExist;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Support\Facades\Gate;
+use Illuminate\Validation\Rule;
+
+class QuestionnaireFrontAnswerRequest extends FormRequest
+{
+    protected function prepareForValidation(): void
+    {
+        parent::prepareForValidation();
+        $this->merge([
+            'id' => $this->route('id'),
+            'model_type_title' => $this->route('model_type_title'),
+            'model_id' => $this->route('model_id')
+        ]);
+    }
+
+    public function authorize(): bool
+    {
+        $questionnaire = $this->getQuestionnaire();
+
+        return Gate::allows('readFront', $questionnaire);
+    }
+
+    public function rules(): array
+    {
+        return [
+            'id' => [
+                'integer',
+                'required',
+                Rule::exists(Questionnaire::class, 'id'),
+            ],
+            'model_type_title' => [
+                'string',
+                Rule::exists(QuestionnaireModelType::class, 'title'),
+                new ClassExist
+            ],
+            'model_id' => [
+                'integer',
+                Rule::exists($this->getQuestionnaireModelType()->model_class, 'id'),
```

Comment:
This will throw exception if model_type_title does not exist in QuestionnaireModelType, so rule for 'model_type_title' will never fail validation as either it is correct or this rule throws 404